crates/guest-rust/src/lib.rs

```diff
@@ -32,6 +32,7 @@ pub mod rt {
             layout = Layout::from_size_align_unchecked(new_len, align);
             alloc::alloc(layout)
         } else {
+            assert_ne!(new_len, 0, "non-zero old_len requires non-zero new_len!");
```

Comment:
It isn't obvious to me whether this should be `debug_assert_ne`.

Reply:
I thought that this was ok, but actually on second thought now that I'm playing around with some modules locally, I think it would be best to keep asserts out of this function. So yeah can you change this to a `debug_assert!`?